packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -153,7 +153,7 @@ protected override ConstructorProvider[] BuildConstructors()
 
             // Build the initialization constructor
             var accessibility = DeclarationModifiers.HasFlag(TypeSignatureModifiers.Abstract)
-                ? MethodSignatureModifiers.Protected
+                ? MethodSignatureModifiers.Private | MethodSignatureModifiers.Protected
```

Comment:
add a test to assert discriminated base has private protected ctor and not just protected